src/Board.hs

```diff
@@ -86,10 +88,10 @@ slideNumber number board =
           validateAndSlideNumber (Just blankPosition) (Just numberPosition)
                 | nextTo blankPosition numberPosition = Right $ swapNumberAndBlank number board
                 | otherwise                           = Left NumberNotAdjacentToBlank
-          nextTo (Position bx by) (Position nx ny)    =
-                (nx == bx && (ny == by-1 || ny == by+1)) ||
-                (ny == by && (nx == bx-1 || nx == bx+1))
-          swapNumberAndBlank number (Board d layout)  = Board d $ map swappy layout
+
+          nextTo (Position bx by) (Position nx ny)    = abs (bx - nx) + abs (by - ny) == 1
```

Comment:
Yours is the correct mathematical solution, and mine is ugly.  I just had an idea for this, although it would be much wordier: I could implement leftOf, rightOf, above, and below in order to get the point across.  Trying that now.